tests/virt_autotest/kubevirt_tests_agent.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: This kubevirt test relies on the upstream test code, which is downstreamed as virt-tests.
+#          This is the part running on agent node.
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base multi_machine_job_base;
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils;
+
+sub run {
+    my ($self) = shift;
+    my $sut_ip = get_required_var('SUT_IP');
+
+    set_var('AGENT_IP', $sut_ip);
+    bmwqemu::save_vars();
+
+    # Synchronize the server & agent nodes
+    # mutex_create('KUBEVIRT_TEST_SETUP');
+    barrier_create('KUBEVIRT_TEST_SETUP', 2);
+    barrier_wait('KUBEVIRT_TEST_SETUP');
+
+    my $server_ip = $self->get_var_from_child("SERVER_IP");
+    record_info('Server IP', $server_ip);
+
+    record_info('Start RKE2 agent setup', '');
+    $self->rke2_agent_setup($server_ip);
+
+    barrier_create('KUBEVIRT_TEST_DONE', 2);
+    barrier_wait('KUBEVIRT_TEST_DONE');
+}
+
+sub rke2_agent_setup {
+    my ($self, $server_ip) = @_;
+
+    systemctl('stop apparmor.service');
+    systemctl('stop firewalld.service');
+    $self->setup_passwordless_ssh_login($server_ip);
+
+    # Ensure SUSE certificates are installed on the node
+    ensure_ca_certificates_suse_installed();
+
+    # Install downstream kubevirt packages
+    zypper_call('in -n kubernetes1.18-client') if (script_run('rpmquery kubernetes1.18-client'));
+
+    # RKE2 deployment on agent node
+    # Default is to setup service with the latest RKE2 version, the parameter INSTALL_RKE2_VERSION allows to setup with a specified version.
+    my $rke2_ver = get_var('INSTALL_RKE2_VERSION');
+    if ($rke2_ver) {
+        record_info('RKE2 version', $rke2_ver);
+        assert_script_run("curl -sfL https://get.rke2.io | INSTALL_RKE2_VERSION=$rke2_ver INSTALL_RKE2_TYPE=agent sh -", timeout => 180);
+    } else {
+        assert_script_run('curl -sfL https://get.rke2.io | INSTALL_RKE2_TYPE=agent sh -', timeout => 180);
+    }
+
+    # Wait for rke2-server service to be ready
+    barrier_create('RKE2_SERVER_START_READY', 2);
+    barrier_wait('RKE2_SERVER_START_READY');
+
+    # Configure rke2-agent service
+    my $server_node_token = script_output("ssh root\@$server_ip cat /var/lib/rancher/rke2/server/node-token");
+    assert_script_run('mkdir -p /etc/rancher/rke2/');
+    assert_script_run("echo 'server: https://$server_ip:9345' > /etc/rancher/rke2/config.yaml");
+    assert_script_run("echo 'token: $server_node_token' >> /etc/rancher/rke2/config.yaml");
+
+    # Enable rke2-agent service
+    systemctl('restart yast-timesync.service');
+    systemctl('enable rke2-agent.service');
+    systemctl('start rke2-agent.service', timeout => 180);
+    check_service_status();
```

Comment:
I think the way you calling member function is not very consistent. Sometimes $self->xxxx, sometimes just check_service_status.

Reply:
Updated.